File: 2138. Divide a String Into Groups of Size k/solution.py
class Solution(object):
  def divideString(self, s, k, fill):
    """
    :type s: str
    :type k: int
    :type fill: str
    :rtype: List[str]
    """
    result = []
    word = ""
    for letter in s:
      word += letter
      if len(word) == k:
        result.append(word)
        word = ""
    if len(word) != 0:
      for index in range(len(word), k):
        word += fill
        if len(word) == k:
          result.append(word)

    return result

File: 2138. Divide a String Into Groups of Size k/solution.js
/**
 * @param {string} s
 * @param {number} k
 * @param {character} fill
 * @return {string[]}
 */
 var divideString = function(s, k, fill) {
  let result = [];
  let word = "";
  for (let index = 0; index < s.length; index++){
    let letter = s[index];
    word += letter;
    if (word.length == k){
      result.push(word);
      word = "";
    }
  }

  if (word.length != 0){
    for (let index = word.length; index < k; index++){
      word += fill;
      if (word.length == k){
        result.push(word);
      }
    }
  }

  return result;
};